docs/conditions.md

```diff
@@ -0,0 +1,150 @@
+`Unknown` for conditions while operators are starting up.
+
+#### OperatorAvailable
+```
+	OperatorAvailable ClusterStatusConditionType = "Available"
+```
+OperatorAvailable indicates that the binary maintained by the operator
+(eg: openshift-apiserver for the openshift-apiserver-operator), is functional
+and available in the cluster.
+
+#### OperatorProgressing
+```
+	OperatorProgressing ClusterStatusConditionType = "Progressing"
+```
+Progressing indicates that the operator is actively making changes to the binary
+maintained by the operator (eg: openshift-apiserver for the
+openshift-apiserver-operator).
+
+#### OperatorDegraded
+```
+	OperatorDegraded ClusterStatusConditionType = "Degraded"
+```
+Degraded indicates that the component operator is not functioning completely.
+An example of a degraded state would be if there should be 5 copies of the
+component running but only 4 are running. It may still be available, but it is
+degraded.
+
+#### Condition Matrix
+
+| Condition        | Status           | Status  | Status  |
+| :------------- |:-------------:|:-----:|:-----:|
+| OperatorAvailable | True | True | True |
+| OperatorProgressing | False | True | True |
+| OperatorDegraded | False | False | True |
+| Meaning | Component is 100% healthy and the Operator is idle | Component is functional but, either upgrading or healing | Component is functioning below capacity and an upgrade or heal is in progress |
+
+| Condition        | Status           | Status  |
+| :------------- |:-------------:|:-----:|
+| OperatorAvailable | False | False |
+| OperatorProgressing | False | True |
+| OperatorDegraded | True | True |
+| Meaning | Component and operator are in a failed state that requires human intervention.  Failed upgrade or failed heal | Component is in a failed state and an operator is healing |
+
+| Condition        | Status           |
+| :------------- |:-------------:|
+| OperatorAvailable | False |
+| OperatorProgressing | True |
+| OperatorDegraded | False |
+| Meaning | Operator is deploying the component |
+
+## Readiness Probe
+With a standardized set of conditions, the HCO should report the health of the
+overall application back to OLM and the user.  This will be critial for sensitive
+operations like upgrade, because OLM needs to know it shouldn't replace an
+operator when it is in the middle of imporant work.
+
+See this [issue](https://github.com/operator-framework/operator-lifecycle-manager/issues/922) for why we only want to report a readiness probe on the HCO
+instead of on all component operators.
+
+## Reason
+`Reason` is _a one-word CamelCase reason for the condition's last transition_.
+
+We'll use a series of lifecycle inspired prefixes paired with postfixes to
+standardize values for `Reason`.
+
+|         | -Failed  | -Succeeded | -Invalid | -InProgress |
+| :------------- |:-------------:|:-----:|:-----:|:-----:|
+| Install- | InstallFailed | InstallSucceeded | InstallInvalid | InstallInProgress |
```

Comment:
It is not clear to me which Reason you would use when transitioning to `Available: true`. I'm not really sure what standardizing on the `Reason` field really buys us at this point. IMO prescribing that operators populate the Reason is sufficient.